packages/eslint-config-godaddy-react-typescript/index.js

```diff
@@ -1,38 +1,43 @@
-module.exports = {
-  extends: [
-    require('./extends')('eslint-config-godaddy-react'),
-    'plugin:eslint-plugin-react/recommended'
-  ],
-  parser: '@typescript-eslint/parser',
-  plugins: ['@typescript-eslint'],
-  overrides: [
-    {
-      files: ['*.js', '*.json', '*.jsx', '*.ts', '*.tsx'],
-      rules: {
-        // note you must disable the base rule as it can report incorrect errors (in typescript)
-        // (because of imported types being incorrectly reported as unused)
-        // eslint-disable-next-line max-len
-        // See https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md
-        'no-unused-vars': 'off',
-        '@typescript-eslint/no-unused-vars': [
-          'error',
-          {
-            vars: 'all',
-            args: 'after-used',
-            ignoreRestSiblings: false
-          }
-        ],
-        // Disable the base rule and enable TypeScript version to avoid incorrect reports
-        // See https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-shadow.md
-        'no-shadow': 'off',
-        '@typescript-eslint/no-shadow': 'warn'
-      }
+import gdReactConfig from 'eslint-config-godaddy-react';
+import tseslintParser from '@typescript-eslint/parser';
+import tseslintPlugin from '@typescript-eslint/eslint-plugin';
+const config = [
+  ...gdReactConfig,
+  {
+    languageOptions: {
+      parser: tseslintParser
     },
-    {
-      files: ['*.ts', '*.tsx'],
-      rules: {
-        'react/prop-types': 'off'
-      }
+    plugins: {
+      '@typescript-eslint': tseslintPlugin
+    },
+    files: ['*.ts', '*.tsx'],
```

Comment:
There are also variations for TypeScript files that produce `.cjs` and `.mjs` files.

```suggestion
    files: ['*.ts', '*.tsx', '*.mts', '*.mtsx', '*.cts', '*.ctsx'],
```